flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/GlobalStreamingCommitterOperatorFactory.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.flink.streaming.runtime.operators.sink;
+
+import org.apache.flink.api.connector.sink.GlobalCommitter;
+import org.apache.flink.api.connector.sink.Sink;
+import org.apache.flink.streaming.api.operators.StreamOperator;
+
+/**
+ * A {@link org.apache.flink.streaming.api.operators.StreamOperatorFactory} for {@link GlobalStreamingCommitterOperator}.
+ *
+ * @param <CommT> The committable type of the {@link GlobalCommitter}.
+ * @param <GlobalCommT> The global committable type of the {@link GlobalCommitter}.
+ */
+public class GlobalStreamingCommitterOperatorFactory<CommT, GlobalCommT> extends AbstractStreamingCommitterOperatorFactory<CommT, GlobalCommT> {
+
+	private final Sink<?, CommT, ?, GlobalCommT> sink;
+
+	public GlobalStreamingCommitterOperatorFactory(Sink<?, CommT, ?, GlobalCommT> sink) {
+		this.sink = sink;
+	}
+
+	@Override
+	AbstractStreamingCommitterOperator<CommT, GlobalCommT> createStreamingCommitterOperator() {
+		return new GlobalStreamingCommitterOperator<>(
+				sink.createGlobalCommitter()
+						.orElseThrow(() -> new IllegalArgumentException(
```

Comment:
Probably throw an `IllegalStateException` instead of an `IllegalArgumentException`. What do you think @guoweiM ?